src/main/java/com/sonymobile/jenkins/plugins/kerberossso/PluginImpl.java

```diff
@@ -127,6 +135,15 @@ private void removeFilter() throws ServletException {
     }
 
     /**
+     * Create and attach the filter.
+     * @throws ServletException Unable to add filter.
+     */
+    private void registerFilter() throws ServletException {
+        this.filter = new KerberosSSOFilter(createConfigMap(), new SpnegoKerberosAuthenticationFactory());
+        PluginServletFilter.addFilter(filter);
```

Comment:
But here you are effectively just adding a new instance to a list of filters already containing older instances with the old config no?